src/server/rpc.rs

```diff
@@ -260,15 +261,18 @@ pub async fn worker_receive_loop<
         match message {
             FromWorkerMessage::TaskFinished(msg) => {
                 on_task_finished(&mut core, &mut *comm, worker_id, msg);
+                insert_worker_message_event(&mut core, FromWorkerMessage::TaskFinished(msg))
```

Comment:
To centralize the event implementation, I suggest to create some structure that will have easy to use methods like `on_task_finished(&mut self, msg: TaskFinishedMsg)`. It would also store the events inside of it and handle all of the event logic. The `Core` would then just store it in an attribute.